DelvEdit/src/com/interrupt/dungeoneer/editor/ui/menu/generator/LevelGeneratorMenuItem.java

```diff
@@ -0,0 +1,129 @@
+
+public class LevelGeneratorMenuItem extends DynamicMenuItem {
+    public LevelGeneratorMenuItem(Skin skin) {
+        super("Generate Level", skin, new DynamicMenuItemAction() {
+            private boolean needsRefresh = false;
+
+            @Override
+            public boolean isDirty() {
+                return needsRefresh;
+            }
+
+            @Override
+            public void updateMenuItem(MenuItem menuItem) {
+                needsRefresh = false;
+
+                if (menuItem.subMenu != null) {
+                    menuItem.subMenu.items.clear();
+                }
+
+                Array<SectionDefinition> sectionDefinitions = Editor.app.generatorInfo.getSectionDefinitions();
+
+                if (sectionDefinitions.size > 0) {
+                    for (SectionDefinition sectionDefinition : sectionDefinitions) {
+                        MenuItem parent = new MenuItem(sectionDefinition.name.toUpperCase(), skin);
+
+                        if (sectionDefinition.levelTemplates.size > 0) {
+                            for (Level template : sectionDefinition.levelTemplates) {
+                                if (template.generated) {
+                                    String label = template.roomGeneratorType + " (" + template.theme + ")";
+                                    parent.addItem(new MenuItem(label, skin, makeLevelGeneratorAction(template)));
+                                } else {
+                                    MenuItem item = new MenuItem("No templates found", skin);
+                                    item.getLabel().setColor(0.5f, 0.5f, 0.5f, 1);
+                                    parent.addItem(item);
+                                }
+                            }
+                        } else {
+                            MenuItem item = new MenuItem("No templates found", skin);
+                            item.getLabel().setColor(0.5f, 0.5f, 0.5f, 1);
+                            parent.addItem(item);
+                        }
+
+                        menuItem.addItem(parent);
+                    }
+
+                    menuItem.addSeparator();
+                    String label = "Re-Generate Level"
+                            + (Editor.app.generatorInfo.lastGeneratedLevelTemplate != null
+                                    ? " (" + Editor.app.generatorInfo.lastGeneratedLevelTemplate.roomGeneratorType
+                                            + ", " + Editor.app.generatorInfo.lastGeneratedLevelTemplate.theme + ")"
+                                    : "");
+                    MenuItem item = new MenuItem(label, skin, makeLevelGeneratorAction())
+                            .setAccelerator(new MenuAccelerator(Keys.L, false, true));
+
+                    if (Editor.app.generatorInfo.lastGeneratedLevelTemplate == null) {
+                        item.getLabel().setColor(0.5f, 0.5f, 0.5f, 1);
+                        // TODO: Change color of accelerator label.
```

Comment:
Same here.

Reply:
That one is more of a general thing that would need a feature implementation, I guess.

I looked at the accelerator menu code, and currently there is no way to also change the color of the accelerator label, as it gets redrawn immediately each tick.

But this is just a visual enhancement for greyed-out options, to get it in line with the menu items label which we are coloring grey.